frontend/tests/playwright/product/add_to_cart.spec.ts

```diff
@@ -80,28 +78,28 @@ test.describe('product page add to cart', () => {
       await cartDrawer.open();
       await cartDrawer.assertCartTotalQuantity(0);
       await cartDrawer.assertItemIsNotPresent(sku);
-      await page.getByTestId('back-to-shopping').click();
+      await productPage.page.getByTestId('back-to-shopping').click();
```

Comment:
Not too sure about this one, but does typescript have some sort of magic methods like php? I'm totally not sure if this is even possible in ts.

In php, we could do something like this on the product page:

```php
public function __call($name, $args) {
   if (<$name isn't a function in this class>) {
     return $this->page->$name($args);
   }
   return $this->$name($args);
}
```

Then this could just be 
```suggestion
      await productPage.getByTestId('back-to-shopping').click();
```

Reply:
I initially was looking into something like for `TS` but I could not find anything.

I'll do some more digging now that I learned they are called `magic methods`.

Reply:
I don't think there is an easy way to implement something like `__call` magic method in Typescript. I saw there is something called a `Proxy` object (https://stackoverflow.com/a/42244695) that could allow us to redirect all implicit methods to `Page` but because of Typescript, there are a lot of type checks we would need to add. In addition, it would convert the methods of the classes into properties.

https://codesandbox.io/s/typescript-playground-export-forked-k3iewl?file=/index.ts

Reply:
Sounds good. I'm good with how this is 👍